src/MongoDB.Driver/FindFluent.cs

```diff
@@ -219,14 +228,51 @@ public override string ToString()
                 sb.Append(".maxTime(" + _options.MaxTime.Value.TotalMilliseconds + ")");
             }
 
+            if (_options.Hint != null)
+            {
+                sb.Append(".hint(" + _options.Hint + ")");
+            }
+
+            if (_options.Max != null)
+            {
+                sb.Append(".max(" + _options.Max + ")");
+            }
+
+            if (_options.MaxScan != null)
+            {
+                sb.Append(".maxScan(" + _options.MaxScan.Value + ")");
+            }
+
+            if (_options.Min != null)
+            {
+                sb.Append(".min(" + _options.Min + ")");
+            }
+
+            if (_options.ReturnKey.HasValue)
+            {
+                sb.Append(".returnKey(" + _options.ReturnKey.Value.ToString().ToLower() + ")");
+            }
+
+            if (_options.ShowRecordId.HasValue)
+            {
+                sb.Append(".showRecordId(" + _options.ShowRecordId.Value.ToString().ToLower() + ")");
+            }
+
+            if (_options.Snapshot != null)
+            {
+                sb.Append(".snapshot(" + _options.Snapshot.Value.ToString().ToLower() + ")");
+            }
+
             if (_options.Comment != null)
             {
                 sb.Append("._addSpecial(\"$comment\", \"" + _options.Comment + "\")");
             }
 
+#pragma warning disable 618
             if (_options.Modifiers != null)
             {
                 foreach (var modifier in _options.Modifiers)
+#pragma warning restore 618
                 {
                     sb.Append("._addSpecial(\"" + modifier.Name + "\", ");
                     if (modifier.Value.BsonType == BsonType.String)
```

Comment:
I'm not sure about `ToString()` implementation:
1. The order of outputting fields seems not to be preserved in the original code, it is not alphabetical and does not follow the order from [find syntax](https://docs.mongodb.com/manual/reference/command/find/index.html#syntax).
2. I'm not sure whether I should use `".fieldName(value)"` or `"._addSpecial($fieldName, value)"` for top level fields: if I use first option for fields listed in [find syntax](https://docs.mongodb.com/manual/reference/command/find/index.html#syntax) and second options for fields not listed in the doc above, then top level fields `MaxScan` and `Snapshot` will likely be confused with `Modifiers` inner fields `$maxScan` and `$snapshot`. That's why I use `".fieldName(value)"` for all top level fields and  `"._addSpecial($fieldName, value)"` for 
 all `Modifiers` inner fields, though existing `Comment` top level field falls out of this approach.
3. I think `ToString()` method shouldn't contain any logic apart from deep listing all object fields, but this way top level fields I added will be listed along with `Modifiers` inner fields. I can change it to contain same logic as for `FindOperation` where top level fields override `Modifiers` inner fields if needed.
4. `ShowRecordId` field is tricky. It is `"$showRecordId"` according to [driver specification](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#operation-parameters), it is also `"showRecordId"` according to [server find syntax](https://docs.mongodb.com/manual/reference/command/find/index.html#syntax), but it is `"$showDiscLoc"` according to [driver ticket](https://jira.mongodb.org/browse/DRIVERS-347) and equivalent to `"$showDiscLoc"` in our code. So my guess is that it should be `ShowRecordId` for top level field and `"$showDiscLoc"` for `Modifiers` inner field. But I'm still not sure which name to choose when translating it to string.

Reply:
This ToString method was already weird before you modified it.

I think your modifications are in the spirit of the existing implementation, so I'm OK with what you've done.